android/src/main/java/com/amplitude/android/utilities/AndroidNetworkConnectivityChecker.kt

```diff
@@ -0,0 +1,49 @@
+package com.amplitude.android.utilities
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import com.amplitude.common.Logger
+
+class AndroidNetworkConnectivityChecker(private val context: Context, private val logger: Logger) {
```

Comment:
Can you please add a test for this class to confirm it works as expected when network state is toggled?

Reply:
Added https://github.com/amplitude/Amplitude-Kotlin/blob/0acb081f3dc5b2b1580905ae5309e1e43815e0b3/android/src/test/java/com/amplitude/android/utilities/AndroidNetworkConnectivityCheckerTest.kt